trace.go

```diff
@@ -0,0 +1,362 @@
+}
+
+// IsSampled returns if the sampling bit is set in the trace flags.
+func (sc SpanContext) IsSampled() bool {
+	return sc.TraceFlags&FlagsSampled == FlagsSampled
+}
+
+type traceContextKeyType int
+
+const (
+	currentSpanKey traceContextKeyType = iota
+	remoteContextKey
+)
+
+// ContextWithSpan returns a copy of parent with span set to current.
+func ContextWithSpan(parent context.Context, span Span) context.Context {
+	return context.WithValue(parent, currentSpanKey, span)
+}
+
+// SpanFromContext returns the current span from ctx, or nil if none set.
+func SpanFromContext(ctx context.Context) Span {
+	if span, ok := ctx.Value(currentSpanKey).(Span); ok {
+		return span
+	}
+	return noopSpan{}
+}
+
+// ContextWithRemoteSpanContext returns a copy of parent with a remote set as
+// the remote span context.
+func ContextWithRemoteSpanContext(parent context.Context, remote SpanContext) context.Context {
+	return context.WithValue(parent, remoteContextKey, remote)
+}
+
+// RemoteSpanContextFromContext returns the remote span context from ctx.
+func RemoteSpanContextFromContext(ctx context.Context) SpanContext {
+	if sc, ok := ctx.Value(remoteContextKey).(SpanContext); ok {
+		return sc
+	}
+	return SpanContext{}
+}
+
+// Span is the individual component of a trace. It represents a single named
+// and timed operation of a workflow that is traced. A Tracer is used to
+// create a Span and it is then up to the operation the Span represents to
+// properly end the Span when the operation itself ends.
+type Span interface {
+	// Tracer returns the Tracer that created the Span. Tracer MUST NOT be
+	// nil.
+	Tracer() Tracer
+
+	// End completes the Span. Updates are not allowed the Span after End is
+	// called other than setting the status.
+	End(options ...SpanOption)
+
+	// AddEvent adds an event to the span.
+	AddEvent(ctx context.Context, name string, attrs ...label.KeyValue)
+	// AddEventWithTimestamp adds an event that occurred at timestamp to the
+	// Span.
+	AddEventWithTimestamp(ctx context.Context, timestamp time.Time, name string, attrs ...label.KeyValue)
+
+	// IsRecording returns the recording state of the Span. It will return
+	// true if the Span is active and events can be recorded.
+	IsRecording() bool
+
+	// RecordError records an error as a Span event.
+	RecordError(ctx context.Context, err error, opts ...ErrorOption)
+
+	// SpanContext returns the SpanContext of the Span. The returned
+	// SpanContext is usable even after the End has been called for the Span.
+	SpanContext() SpanContext
+
+	// SetStatus sets the status of the Span in the form of a code and a
+	// message. SetStatus overrides the value of previous calls to SetStatus
+	// on the Span.
+	SetStatus(code codes.Code, msg string)
+
+	// SetName sets the Span name.
+	SetName(name string)
+
+	// SetAttributes sets kv as attributes of the Span. If the keys of kv
+	// already exist for an attribute of the Span it will be overwritten with
+	// the value contained in kv.
```

Comment:
```suggestion
	// SetAttributes sets kv as attributes of the Span. If a key frm kv
	// already exists for an attribute of the Span it will be overwritten with
	// the value contained in kv.
```